src/main/groovy/com/twcable/grabbit/jcr/JCRNodeDecorator.groovy

```diff
@@ -105,6 +106,37 @@ class JCRNodeDecorator {
         return definition.isMandatory()
     }
 
+    void checkoutIfNecessary(javax.jcr.Node node) throws RepositoryException {
```

Comment:
Why are we passing in a node? This is a decorator. We are delegating to a wrapped node (See line 39)